spark/src/main/java/org/opensearch/sql/spark/storage/SparkStorageFactory.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.spark.storage;
+
+import java.util.Map;
+import lombok.RequiredArgsConstructor;
+import org.opensearch.client.Client;
+import org.opensearch.sql.common.setting.Settings;
+import org.opensearch.sql.datasource.model.DataSource;
+import org.opensearch.sql.datasource.model.DataSourceMetadata;
+import org.opensearch.sql.datasource.model.DataSourceType;
+import org.opensearch.sql.spark.client.SparkClient;
+import org.opensearch.sql.storage.DataSourceFactory;
+import org.opensearch.sql.storage.StorageEngine;
+
+/**
+ * Storage factory implementation for spark connector.
+ */
+@RequiredArgsConstructor
+public class SparkStorageFactory implements DataSourceFactory {
+  private final Client client;
```

Comment:
why depend on opensearch client?

Reply:
to fetch query result from `.query_execution_result` index